app/controllers/dish_ingredients_controller.rb

```diff
@@ -0,0 +1,13 @@
+class DishIngredientsController < ApplicationController
+  def create
+    @dish = Dish.find(params[:dish_id])
+    ingredient = Ingredient.find(dish_ingredient_params[:ingredient_id])
+    
+    DishIngredient.create(dish_id: params[:dish_id], ingredient_id: ingredient.id)
+    redirect_to dish_path(params[:dish_id])
+  end
+
+  def dish_ingredient_params
```

Comment:
I'm not sure why you used this params method on line 4, here I would have just done `Ingredient.find(params[:dish_ingredient][:ingredient_id])` .... though I guess I don't love that. I think because of how your form is setup, it makes querying for the params here a bit awkward. Great job getting it all to work!